packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -0,0 +1,440 @@
+    };
+  }
+
+  /**
+   * Creates spans for outgoing requests, sending spans' context for distributed
+   * tracing.
+   */
+  protected _getPatchOutgoingRequestFunction() {
+    return (original: Func<ClientRequest>): Func<ClientRequest> => {
+      return this.outgoingRequestFunction(original, this);
+    };
+  }
+
+  protected _getPatchOutgoingGetFunction() {
+    return (original: Func<ClientRequest>): Func<ClientRequest> => {
+      // Re-implement http.get. This needs to be done (instead of using
+      // getPatchOutgoingRequestFunction to patch it) because we need to
+      // set the trace context header before the returned ClientRequest is
+      // ended. The Node.js docs state that the only differences between
+      // request and get are that (1) get defaults to the HTTP GET method and
+      // (2) the returned request object is ended immediately. The former is
+      // already true (at least in supported Node versions up to v10), so we
+      // simply follow the latter. Ref:
+      // https://nodejs.org/dist/latest/docs/api/http.html#http_http_get_options_callback
+      // https://github.com/googleapis/cloud-trace-nodejs/blob/master/src/plugins/plugin-http.ts#L198
+      return function outgoingGetRequest(options: string | RequestOptions | URL, callback?: HttpCallback) {
+        const req = request(options, callback);
+        req.end();
+        return req;
+      };
+    };
+  }
+
+  /**
+   * Injects span's context to header for distributed tracing and finshes the
+   * span when the response is finished.
+   * @param original The original patched function.
+   * @param options The arguments to the original function.
+   */
+  private getMakeRequestTraceFunction(
+    request: ClientRequest,
+    options: RequestOptions,
+    plugin: HttpPlugin
+  ): Func<ClientRequest> {
+    return (span: Span): ClientRequest => {
+      plugin._logger.debug('makeRequestTrace');
+
+      if (!span) {
+        plugin._logger.debug('makeRequestTrace span is null');
+        return request;
+      }
+
+      const propagation = plugin._tracer.getHttpTextFormat();
+      // If outgoing request headers contain the "Expect" header, the returned
+      // ClientRequest will throw an error if any new headers are added.
+      // So we need to clone the options object to be able to inject new
+      // header.
+      if (HttpPlugin.hasExpectHeader(options)) {
+        options = Object.assign({}, options);
+        options.headers = Object.assign({}, options.headers);
+      }
+      propagation.inject(span.context(), HttpPlugin.PROPAGATION_FORMAT, options.headers);
+
+      request.on('response', (response: IncomingMessage) => {
+        plugin._tracer.wrapEmitter(response);
+        plugin._logger.debug('outgoingRequest on response()');
+        response.on('end', () => {
+          plugin._logger.debug('outgoingRequest on end()');
+          const method = response.method ? response.method.toUpperCase() : 'GET';
```

Comment:
`const method = (response.method || 'GET').toUpperCase()`

Reply:
Are you sure on this one ? make upper case on `GET` ?
Can you explain me why you prefer `const method = (response.method || 'GET').toUpperCase()` ?
Thanks in advance.

Reply:
Looks smaller and nit, calling `toUpperCase()` on `GET` won't add big overhead. Having said that, I am fine with the current approach.